influxdb3_server/src/http.rs

```diff
@@ -477,6 +499,39 @@ where
         Ok(())
     }
 
+    async fn extract_query_request<D: DeserializeOwned>(
+        &self,
+        req: Request<Body>,
+    ) -> Result<QueryRequest<D, QueryFormat, StatementParams>> {
+        let header_format = QueryFormat::try_from_headers(req.headers())?;
+        let request = match *req.method() {
+            Method::GET => {
+                let query = req.uri().query().ok_or(Error::MissingQueryParams)?;
+                let r = serde_urlencoded::from_str::<QueryRequest<D, Option<QueryFormat>, String>>(
+                    query,
+                )?;
+                QueryRequest {
+                    database: r.database,
+                    query_str: r.query_str,
+                    format: r.format,
+                    params: r.params.map(|s| serde_json::from_str(&s)).transpose()?,
+                }
+            }
+            Method::POST => {
+                let body = self.read_body(req).await?;
+                serde_json::from_slice(body.as_ref())?
+            }
+            _ => todo!(),
```

Comment:
Did you mean to make this unreachable or did you mean to handle this branch?

Reply:
Thanks for catching. I added the error variant for this then forgot to use it 🤦